### PR TITLE
Prevent random Pug fleets from leaving Pug systems

### LINKED ISSUE #8876: Pug fleets shouldn't leave their systems and jump into human or Wanderer space
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Currently, a fleet, as long as it spawned, would jump randomly to a nearby system. This create inconsistencies, as this causes fleets to jump in areas they are not supposed to be in.

For instance, this can cause Pugs from Deneb to jump in Republic space, and even land on their planets.

Also, by waiting a very long time in `Pug Iyik`, fleets of out-of-fuel pugs will accumulate in `Sko'karak`, and so this also causes stranded Pug parties.

### Steps to Reproduce

This can be reproduced in human space by going to Deneb, and then jumping to a nearby system after some pugs spawned. But the low Pug spawning rate here, and the amount of systems they can possibly go to, make it harder to observe, so here is an alternative using the Wanderer Pugs:

1. Load the save provided below.
2. Jump north to `Sko'karak`.
3. Spend some time here to observe the normal behavior: No Pugs spawn here.
4. Jump west to `Pug Iyik`.
5. Wait a little so that Pugs fleets take off from the planet.
6. Jump back to `Sko'karak`.
7. Wait to observe the Pug fleets jumping to `Sko'karak`. If they do not arrive, repeat point 4.

### Expected Behavior

Pugs should not jump to `Sko'karak`, beacuse they are not supposed to have fleets here.
Fleets should not jump to systems they should not be able to spawn in.

Perhaps a fleet destination could be determined randomly from the systems they are actually able to spawn in? The destination may be chosen randomly from those. The weight/chances of a destination being chosen may even depend on the spawn rate of this fleet at that location. Fleets may also have a chance to choose to land instead. For fleets equipped with jump drives, alternatively (I am thinking of the pugs), they might very well be jumping to an unknown invisible unmapped system (in theory, jumping ships may simply disappear).

### Screenshots

image

### Link to save file

The save:
```
pilot Destination Deneb
date 16 11 3013
"system entry method" takeoff
system Ka'ch'chrai
planet "Vara K'chrai"
clearance
playtime 45.798703
"flagship index" 0
"map coloring" -6
"map zoom" 0
ship Shuttle
	name "My Ship"
	sprite ship/shuttle
		"frame rate" 15
		delay 14
	thumbnail thumbnail/shuttle
	uuid 8f2dba08-a2b0-4ef6-98cf-9dec3db6cca8
	attributes
		category Transport
		cost 180000
		mass 70
		bunks 6
		"cargo space" 20
		drag 1.7
		"engine capacity" 60
		"fuel capacity" 160000
		"gun ports" 1
		"heat dissipation" 0.8
		hull 600000
		"outfit space" 120
		"required crew" 1
		shields 500000
		"weapon capacity" 10
	outfits
		"Jump Drive"
		"LP036a Battery Pack"
		"X2200 Ion Steering"
		"X2700 Ion Thruster"
		"nGVF-AA Fuel Cell"
	crew 1
	fuel 160000
	shields 500000
	hull 600000
	position 0 0
	engine -6 30
		zoom 1
		angle 0
		under
	engine 6 30
		zoom 1
		angle 0
		under
	gun 0 -31
		under
	leak leak 60 50
	explode "small explosion" 5
	explode "tiny explosion" 10
	system Ka'ch'chrai
	planet "Vara K'chrai"
licenses
	Pilot's
account
	credits 161000000
	score 400
visited Deneb
visited Ka'ch'chrai
visited "Pug Iyik"
visited Rasalhague
visited Rutilicus
visited Sko'karak
visited Sol
"visited planet" "New Boston"
"visited planet" "Vara Pug"
start default
	system Rutilicus
	planet "New Boston"
	date 16 11 3013
	account
		credits 480000
		score 400
		history
		mortgage Mortgage
			principal 480000
			interest 0.004
			term 365
```

### Operating System

Ubuntu

### Game Source

GitHub Releases

### Game Version

v0.10.1

### Additional Information

# Related Issues
- #6864
- #5329

### PR DESCRIPTION
**Pull request by Amazinite**
**Content (Artwork / Missions / Jobs)**

## Summary
Prevent random Pug fleets from leaving Pug systems and jumping into human or Wanderer space by giving the two Pug governments travel restrictions to not travel to any location that isn't one of their own. In order to not have Pug fleets stack up in single systems and still appear to jump in and out, added three hidden "pocket dimension" systems that share a position with the main systems and act as viable jump destinations for Pug fleets.

No changes made to campaign Pug fleets because NPCs don't follow travel restrictions by default.

Resolves #8876.

## Testing done

Observed that Pug fleets still spawn in Pug systems. They sometimes jump out, but they never jumped out to an accessible system, meaning they're jumping to the hidden pocket dimension systems.

### CODE REVIEW

**Comment by Anarchist2** [comment]
What about using a new government for the pocket dimensions, and restricting the Pug to only be able to travel to those systems (i.e., not their home systems)? That way, you only need to make one system to stop the Pug from jumping in and out.

**Comment by Amazinite** [comment]
There only needs to be one system anyway. I added three systems so that the Pug that jump out aren't constantly jumping in and out of the same system; they might be bouncing between the hidden systems.

**Comment by Amazinite** [comment]
That way they only ever jump into the system to spawn and out of the system to leave. This way they sometimes land on or take off from the planet in the system.

**Comment by Amazinite** [comment]
That method also means that during the FW campaign, the Pug would never travel between any of the systems in their expanded territory.

File: data/map systems.txt [hunk 1]
```diff
@@ -26743,6 +26758,22 @@ system "Pug Iyik"
...
+	hidden
+	pos -361.149 -883.233
+	government "Pug (Wanderer)"
+
+system "Pug Iyik PD 3"
+	hidden
+	pos -361.149 -883.233
+	government "Pug (Wanderer)"
+
+
```
Comment (mOctave) [inline data/map systems.txt:26776]: Is this extra newline intentional?

Comment (Amazinite) [reply data/map systems.txt:26776]: Nope.
```suggestion
```

**Review by Hurleveur — APPROVED** [review]
just tested it, looks like it works well

File: data/map systems.txt [hunk 2]
```diff
@@ -26743,6 +26758,21 @@ system "Pug Iyik"
 			distance 365
 			period 36.706
 
+system "Pug Iyik PD 1"
+	hidden
+	pos -361.149 -883.233
+	government "Pug (Wanderer)"
```
Comment (ziproot) [inline data/map systems.txt:26764]:
```suggestion
	government "Pug (Wanderer)"
	attributes "inaccessible"
```

File: data/map systems.txt [hunk 3]
```diff
@@ -26743,6 +26758,21 @@ system "Pug Iyik"
...
+
+system "Pug Iyik PD 2"
+	hidden
+	pos -361.149 -883.233
+	government "Pug (Wanderer)"
```
Comment (ziproot) [inline data/map systems.txt:26769]:
```suggestion
	government "Pug (Wanderer)"
	attributes "inaccessible"
```

File: data/map systems.txt [hunk 4]
```diff
@@ -26743,6 +26758,21 @@ system "Pug Iyik"
...
+
+system "Pug Iyik PD 3"
+	hidden
+	pos -361.149 -883.233
+	government "Pug (Wanderer)"
```
Comment (ziproot) [inline data/map systems.txt:26774]:
```suggestion
	government "Pug (Wanderer)"
	attributes "inaccessible"
```

File: data/map systems.txt [hunk 5]
```diff
@@ -10244,6 +10244,21 @@ system Deneb
 		distance 1788.4
 		period 1210.08
 
+system "Deneb PD 1"
+	hidden
+	pos -348 225
+	government Pug
```
Comment (ziproot) [inline data/map systems.txt:10250]:
```suggestion
	government Pug
	attributes "inaccessible"
```

File: data/map systems.txt [hunk 6]
```diff
@@ -10244,6 +10244,21 @@ system Deneb
...
+
+system "Deneb PD 2"
+	hidden
+	pos -348 225
+	government Pug
```
Comment (ziproot) [inline data/map systems.txt:10255]:
```suggestion
	government Pug
	attributes "inaccessible"
```

File: data/map systems.txt [hunk 7]
```diff
@@ -10244,6 +10244,21 @@ system Deneb
...
+
+system "Deneb PD 3"
+	hidden
+	pos -348 225
+	government Pug
```
Comment (ziproot) [inline data/map systems.txt:10260]:
```suggestion
	government Pug
	attributes "inaccessible"
```

**Review by ziproot — CHANGES_REQUESTED** [review]
From what I can tell, it is impossible to check for whether or not a system is hidden inside missions, and hidden systems can be chosen by missions. The remnant research mission at the moment would be able to pick these systems. These changes make it so the remnant research mission is unable to pick these systems.

**Comment by Hurleveur** [comment]
It'd be fun if the remnant point you to these systems actually but you're right that they'd have to be more developped first

**Review by ziproot — APPROVED** [review]
This looks good now.

**Comment by mOctave** [comment]
Just to make sure, they can be `inaccessible` and still be accessed by NPCs, right?

**Comment by Amazinite** [comment]
Inaccessible is just an attribute that only influences the locations that missions/jobs can choose from.

**Pull request merged**